Hi @sgugger, I removed bunch of unused imports since it's guaranteed that process group will be initialized by `training_args_sm.py`.

Could you give some pointers on how to fix CI errors?

`F401 'smdistributed.dataparallel.torch.torch_smddp' imported but unused`
this import is needed to register SMDDP as a pytorch backend, can we disable style check on this line?

`FAILED tests/funnel/test_modeling_funnel.py::FunnelModelTest::test_pt_tf_model_equivalence`

--- a/src/transformers/sagemaker/training_args_sm.py
+++ b/src/transformers/sagemaker/training_args_sm.py
@@ -85,7 +85,7 @@ class SageMakerTrainingArguments(TrainingArguments):
             device = torch.device("cuda", local_rank)
             self._n_gpu = 1
         elif is_sagemaker_dp_enabled():
-            import smdistributed.dataparallel.torch.torch_smddp # noqa: F401
+            import smdistributed.dataparallel.torch.torch_smddp  # noqa: F401
 
             torch.distributed.init_process_group(backend="smddp")
             self.local_rank = int(os.getenv("SMDATAPARALLEL_LOCAL_RANK"))

--- a/src/transformers/sagemaker/training_args_sm.py
+++ b/src/transformers/sagemaker/training_args_sm.py
@@ -85,7 +85,7 @@ class SageMakerTrainingArguments(TrainingArguments):
             device = torch.device("cuda", local_rank)
             self._n_gpu = 1
         elif is_sagemaker_dp_enabled():
-            import smdistributed.dataparallel.torch.torch_smddp
+            import smdistributed.dataparallel.torch.torch_smddp # noqa: F401
 
             torch.distributed.init_process_group(backend="smddp")
             self.local_rank = int(os.getenv("SMDATAPARALLEL_LOCAL_RANK"))

